src/System.Management.Automation/engine/PSConfiguration.cs

```diff
@@ -548,6 +548,7 @@ internal sealed class PowerShellPolicies
         public ProtectedEventLogging ProtectedEventLogging { get; set; }
         public Transcription Transcription { get; set; }
         public UpdatableHelp UpdatableHelp { get; set; }
+        public UpdatableHelp PipelineMaxStackSizeMB { get; set; }
```

Comment:
The `PipelineMaxStackSizeMB` setting is not a policy in windows powershell, and thus it shouldn't be added to `PowerShellPolicies`. It should be a setting like `psmodulepath`, or `executionpolicy` or `ConsolePrompting`.

THe `PipelineMaxStackSizeMB` was stored at `SOFTWARE\Microsoft\PowerShell\1\ShellIds` in windows powershell, not at `SOFTWARE\Policies\...`

Reply:
I believe [previously](https://github.com/PowerShell/PowerShell/issues/5809#issuecomment-356059863) we agreed that PowerShell Core is a separate product and it should have its policies and not use policies and configs of Windows PowerShell.
Why don't we have consistency between the settings in the file and the registry if we the consistency by default on Unix? I wouldn't be surprised if MSFT decides to port Registry API to make it easier to port applications. (I would take this into account).
Also if the settings are in different places, it is difficult to find it and makes it very difficult to configure.
Another thought is that we must put (later?) config options and  GPO policies in different registry keys.

I think we should develop this my [comment](https://github.com/PowerShell/PowerShell/issues/5809#issuecomment-355843863) about priority - maybe open new tracking Issue?
Also we could open new tracking Issue (?) to enumerate config options and to solve where put different options - in policy key or/and in config key.

Today I suggest:
1. Have `PipelineMaxStackSizeMB` (and most other options) in registry and file.
2. Put `PipelineMaxStackSizeMB` in "Software\Policies\Microsoft\PowerShellCore". Later we can move/duplicate it in "Software\Microsoft\PowerShellCore" - today we haven't code to process the registry branches in priority.

Reply:
I didn't mean `PipelineMaxStackSizeMB` should be put in Registry on Windows. I think this setting should be in the config file only on both Windows and Unix.

As far as I understand, only the GPO related settings will live in Registry and `powershell.config.json` on Windows for PowerShell Core (registry setting takes precedence). Non-GPO settings that used to be stored in Registry in Windows PowerShell now should live in the config file only in PowerShell Core, such as `executionpolicy`, `ConsolePrompting` and many others. You can take a look at `PSConfiguration.cs` to find more of them.

The type `PowerShellPolicies` only represents the GPO settings. `PipelineMaxStackSizeMB` is not a policy setting but a normal powershell setting, so it shouldn't be put in `PowerShellPolicies`.

Reply:
@daxian-dbw I wanted to create Issue but the text was too big and I made a document #5947